src/DataCollection.Shared/ViewModels/AuthViewModel.cs

```diff
@@ -110,15 +110,20 @@ public ICommand SignOutCommand
             get
             {
                 return _signOutCommand ?? (_signOutCommand = new DelegateCommand(
-                    (x) =>
+                    async (x) =>
                     {
                         // Close any popups
                         BroadcastMessenger.Instance.RaiseBroadcastMessengerValueChanged(null, BroadcastMessageKey.ClosePopups);
 
-                        // clear credentials
-                        foreach (var credential in AuthenticationManager.Current.Credentials)
+                        try 
+                        { 
+                            // Revoke and remove all credentials
+                            await AuthenticationManager.Current.RemoveAndRevokeAllCredentialsAsync();
+                        }
+                        catch (Exception)
                         {
-                            AuthenticationManager.Current.RemoveCredential(credential);
+                            // If revocation fails (e.g. because of network issues), just delete the credentials
+                            AuthenticationManager.Current.RemoveAllCredentials();
```

Comment:
Just had another thought here. Can you confirm this line is necessary? If credentials are removed before they are revoked, wouldn't they be removed regardless of whether the async revoke request was successful?